src/Raven.Server/Documents/Indexes/CollectionOfBloomFilters.cs

```diff
@@ -201,6 +211,16 @@ public void Flush()
             if (_filters == null || _filters.Length == 0)
                 return;
 
+            if (Consumed)
```

Comment:
How does it behave after reloading the index?

Reply:
after index reset the CollectionOfBloomFilters is recreated, and repopulated with BloomFilters

Reply:
I'm talking here more about shutting down the database, not resetting the index.

Reply:
If it was consumed then it will be marked as consumed from the storage when DB reset

Reply:
Great, can you write a test for this?

Reply:
Sure !